python/tests/test_topology.py

```diff
@@ -97,6 +98,618 @@ def generate_segments(n, sequence_length=100, seed=None):
     return segs
 
 
+def kc_distance_tree(tree1, tree2, lambda_param=0):
+    """
+    Returns the Kendall-Colijn distance between the specified pair of trees.
+    lambda_param determines weight of topology vs branch lengths in calculating
+    the distance. Set lambda_param at 0 to only consider topology, set at 1 to
+    only consider branch lengths. See Kendall & Colijn (2016):
+    https://academic.oup.com/mbe/article/33/10/2735/2925548
+    """
+    samples = tree1.tree_sequence.samples()
+    if not np.array_equal(samples, tree2.tree_sequence.samples()):
+        raise ValueError("Trees must have the same samples")
+    if not len(tree1.roots) == len(tree2.roots) == 1:
+        raise ValueError("Trees must have one root")
+    k = samples.shape[0]
+    n = (k * (k - 1)) // 2
+    m = [np.ones(n + k), np.ones(n + k)]
+    M = [np.zeros(n + k), np.zeros(n + k)]
+    for tree_index, tree in enumerate([tree1, tree2]):
+        stack = [(tree.root, 0, tree.time(tree.root))]
+        while len(stack) > 0:
+            u, depth, time = stack.pop()
+            children = tree.children(u)
+            for v in children:
```

Comment:
huh, this is clever, doing a DFS from the root. i might give this a try as well.